invoke-ai/InvokeAI
# feat(mm): remove autoimport, revised startup model scanning

## Summary

Copying the meat-and-potatoes from commit messages. There are a few supporting changes (docs, frontend) per commits.

[feat(mm): remove autoimport; revise startup model scanning](https://github.com/invoke-ai/InvokeAI/commit/635a60e8e09c603c0adc6e86e89b6b35b33985d2)

These two changes are interrelated.

## Autoimport

The autoimport feature can be easily replicated using the scan folder tab in the model manager. Removing the implicit autoimport reduces surface area and unifies all model installation into the UI.

This functionality is removed, and the `autoimport_dir` config setting is removed.

## Startup model dir scanning

We scanned the invoke-managed models dir on startup and took certain actions:

- Register orphaned model files
- Remove model records from the db when the model path doesn't exist

### Orphaned model files

We should never have orphaned model files during normal use - we manage the models directory, and we only delete files when the user requests it.

During testing or development, when a fresh DB or memory DB is used, we could end up with orphaned models that should be registered.

Instead of always scanning for orphaned models and registering them, we now only do the scan if the new `scan_models_on_startup` config flag is set.

The description for this setting indicates it is intended for use for testing only.

### Remove records for missing model files

This functionality could unexpectedly wipe models from the db.

For example, if your models dir was on external media, and that media was inaccessible during startup, the scan would see all your models as missing and delete them from the db.

The "proactive" scan is removed. Instead, we will scan for missing models and log a warning if we find a model whose path doesn't exist. No possibility for data loss.

## Related Issues / Discussions

Discord thread: https://discord.com/channels/1020123559063990373/1149513695567810630/1222097965045514250

## QA Instructions

Test the revised startup model scan:
- Set `use_memory_db: true` in `invokeai.yaml`
- Start up the app
- You should have no models
- Set `scan_models_on_startup: true` in `invokeai.yaml`
- Start up the app
- It should register all your models

Test orphaned model change:
- Move a model out of your models dir
- Start the app
- You should get a warning in terminal saying the model is missing, but it should still be in the model manager. Generation should fail if you try with it
- Move the model back into the models dir
- Start the app
- Should have no warning, and the model should work
 
## Merge Plan

N/A

## Checklist
